distributed/comm/ucx.py

```diff
@@ -60,8 +68,35 @@ def init_once():
                 "CUDA support with UCX requires Numba for context management"
             )
 
+        cuda_visible_device = int(
+            os.environ.get("CUDA_VISIBLE_DEVICES", "0").split(",")[0]
+        )
+        pre_existing_cuda_context = has_cuda_context()
+        if pre_existing_cuda_context is not False:
+            warnings.warn(
+                f"A CUDA context for device {pre_existing_cuda_context} already exists on process "
+                f"ID {os.getpid()}. This is often the result of a CUDA-enabled library calling a "
+                "CUDA runtime function before Dask-CUDA can spawn worker processes. Please make "
+                "sure any such function calls don't happen at import time or in the global scope "
+                "of a program."
+            )
+
         numba.cuda.current_context()
 
+        cuda_context_created = has_cuda_context()
```

Comment:
Same question as above

Reply:
Same as above, I don't see where the potential recursiveness you're asking about would lie.